esp-config/src/generate.rs

```diff
@@ -472,35 +215,64 @@ fn env_change_work_around<W: Write>(stdout: &mut W) {
     }
 }
 
-fn create_config<W: Write>(
-    stdout: &mut W,
+/// A configuration option.
+#[derive(Serialize)]
+pub struct ConfigOption {
```

Comment:
definitely nicer than using a tuple

we could even add convenience ctors for common things if we want